### PR TITLE
Feature: Entity availability check

### PR DESCRIPTION
**Pull request by Tony763**
- Added routine for entity availability check
- Address PR #38

## Checklist:
  - [ ] Code is Python 3.x compatible
  - [ ] Travis is passing tests **Your PR cannot be merged unless tests pass**
  - [ ] No PEP Bot Errors or Issues

### CODE REVIEW

**Comment by stratus-ss** [comment]
@Tony763 this one has a merge conflict. It's an easy fix

**Comment by Tony763** [comment]
Fixed

**Comment by Tony763** [comment]
@stratus-ss when will you have a time, check this one first so I can continue with next PRs.

**Comment by stratus-ss** [comment]
This one LGTM but needs to pend until I can test it. 

I don't *see* a problem in the code, it makes logical sense. I just need to set aside some time to test this. Probably next week

**Comment by Tony763** [comment]
Ok, I will wait with next batch of PRs.

**Comment by stratus-ss** [comment]
Haven't forgotten about this one. Just havent had time to test it yet

**Comment by renekliment** [comment]
Hi! Is there anything that can be done for example by the community to push this forward? 
Are there test cases that we could try and report back?
Thanks!

**Comment by stratus-ss** [comment]
> Hi! Is there anything that can be done for example by the community to push this forward?
> Are there test cases that we could try and report back?
> Thanks!

Tests are always helpful! As are exact examples of how one might test a specific PR.

i.e. when you run the following commands, you should expect the following output

File: __init__.py [hunk 1]
```diff
@@ -109,6 +109,15 @@ def _find_entity(self, entity, domains):
                               "dev_name": entity})
         return ha_entity
 
+    # Routine for entiti availibility check
+    def _check_availability(self, ha_entity):
+        self.log.debug(ha_entity['state'])
```
Comment (stratus-ss) [inline __init__.py]: probably should remove debug code

File: __init__.py [hunk 2]
```diff
@@ -109,6 +109,15 @@ def _find_entity(self, entity, domains):
                               "dev_name": entity})
         return ha_entity
 
+    # Routine for entiti availibility check
+    def _check_availability(self, ha_entity):
```
Comment (stratus-ss) [inline __init__.py:132]: in the future a doc string would be helpful for reviewers to understand the method at a glance

File: __init__.py (hunk 1 above)
Comment (Tony763) [reply __init__.py]: Done

File: __init__.py (hunk 2 above)
Comment (Tony763) [reply __init__.py:132]: Done

Comment (Tony763) [reply __init__.py:132]: Done

File: __init__.py [hunk 3]
```diff
@@ -109,6 +109,20 @@ def _find_entity(self, entity, domains):
...
+        """ Simple routine for checking availability of entity inside
+        Home Assistent. """
+
+        if ha_entity['state'] == 'unavailable':
+            """ Check if state is `unavailable`, if yes, inform user about it. """
+
+            self.speak_dialog('homeassistant.device.unavailable', data={
+                            "dev_name": ha_entity['dev_name']})
+       """ Return result to underliing function. """
```
Comment (stratus-ss) [inline __init__.py]: This line needs to match the indent of the IF statement. It throws an indentation error

**Comment by stratus-ss** [comment]
I had to adjust the following lines

```
-from mycroft import MycroftSkill, intent_file_handler
+from mycroft import MycroftSkill, intent_file_handler, intent_handler
``` 

```
-       """ Return result to underliing function. """
+            """ Return result to underliing function. """
```

**Comment by Tony763** [comment]
@stratus-ss: Sorry, stupid copy paste error. Resolved. Let me know if it is ok, I will rebase changes.

**Review by stratus-ss — APPROVED** [review]
LGTM thanks for the update.

We made a few videos about this PR 
https://www.youtube.com/watch?v=156h-b_t8R8

That's largely what held this up. Thanks for your patience

**Pull request merged**